### PR TITLE
Lacrymology/waitless melted node

### PR DESCRIPTION
**Pull request by Lacrymology**
This works (passes tests) together with the melted-node version of https://github.com/inaes-tic/melted-node/pull/6

We should probably bump versions in both packages

### CODE REVIEW

**Comment by xaiki** [comment]
please do the version bumps, 
we don't review until travis pases.

**Comment by Lacrymology** [comment]
OK, but https://github.com/inaes-tic/melted-node/pull/6 needs to be merged, then

**Comment by xaiki** [comment]
Have jmr review it.

## Or point your build no it.

Niv Sardi, mobile.

**Comment by jmrunge** [comment]
I've reviewed melted-node version and pointed out just two things.  Please
@Lacrymology answer me there so I can merge it and got ahead with reviewing
this PR.

2013/9/7 Niv Sardi notifications@github.com

> Have jmr review it.
> 
> ## Or point your build no it.
> 
> Niv Sardi, mobile.

File: drivers/mvcp/melted-node-driver.js [hunk 1]
```diff
@@ -1,33 +1,41 @@
...
+    logger.debug(this.uuid + " - Creating server instance [" + host + ":" + port + "]");
     this.mlt = new melted_node(host, port, melted_log, timeout);
-    logger.debug(self.uuid + " - Server instance created [" + this.mlt.host + ":" + this.mlt.port + "]");
+    this.commandQueue = Q.resolve();
+    logger.debug(this.uuid + " - Server instance created [" + this.mlt.host + ":" + this.mlt.port + "]");
 }
 
+melted.prototype._sendCommand = function(command) {
+    logger.debug(this.uuid + " - Sending command: " + command);
+    return this.mlt.sendCommand(command, "200 OK");
```
Comment (jmrunge) [inline drivers/mvcp/melted-node-driver.js]: Please remove unnecessary "expected response" parameter as melted-node wont use it anymore

File: drivers/mvcp/melted-node-driver.js [hunk 2]
```diff
@@ -1,33 +1,41 @@
...
+        ret.resolve(self._sendCommand(command));
+    });
+    return ret.promise;
 };
 
 melted.prototype.getServerPlaylist = function() {
     var self = this;
-    logger.debug(self.uuid + " - Sending command: LIST U0");
-    return self.mlt.sendPromisedCommand("list u0", "201 OK").then(function(response) {
+    return this.sendCommand("list u0", "201 OK").then(function(response) {
```
Comment (jmrunge) [inline drivers/mvcp/melted-node-driver.js]: this.sendCommand receives only a command as parameter

File: drivers/mvcp/melted-node-driver.js [hunk 3]
```diff
@@ -68,8 +76,7 @@ melted.prototype.getServerPlaylist = function() {
 
 melted.prototype.getServerStatus = function() {
     var self = this;
-    logger.debug(self.uuid + " - Sending command: USTA U0");
-    return self.mlt.sendPromisedCommand("usta u0", "202 OK").then(function(response) {
+    return this.sendCommand("usta u0", "202 OK").then(function(response) {
```
Comment (jmrunge) [inline drivers/mvcp/melted-node-driver.js]: Idem previous comment

File: drivers/mvcp/melted-node-driver.js [hunk 4]
```diff
@@ -115,21 +122,19 @@ melted.prototype.getServerStatus = function() {
...
 
     var deferred = Q.defer();
 
-    var result = self.mlt.connect();
+    var result = this.mlt.connect();
 
     result.then(function(response) {
-        logger.debug(self.uuid + " - Sending command: ULS");
-        var aux = self.mlt.sendPromisedCommand("ULS", "201 OK");
+        var aux = self.sendCommand("ULS", "201 OK");
```
Comment (jmrunge) [inline drivers/mvcp/melted-node-driver.js]: Idem previous comment

File: test/mvcp-driver-test.js [hunk 5]
```diff
@@ -11,737 +13,783 @@ describe('start mvcp-driver test', function(done) {
...
-        it('--Status: Clip playlist id: should return 1', function() {
-            assert.equal(st.currentClip.playlistId, 1);
-        });
-        it('--Status: Clip id: should return 1', function() {
-            assert.equal(st.currentClip.id, 1);
-        });
-        it('--Status: Unit status: should return playing', function() {
-            assert.equal(st.status, "playing");
+        it('-- mvcp server created', function(done) {
+            server = silence(function(){ return mvcp_server("melted"); });
```
Comment (jmrunge) [inline test/mvcp-driver-test.js]: Please remove call to "silence" function since you are doing such a refactor :)

**Pull request merged**